### ISSUE TITLE
bug: can't build docker image with NextAuth.js

### ISSUE THREAD

**Issue by vojtechmares** [t3-oss/create-t3-app#992]
### Provide environment information

  System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 111.53 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 19.3.0 - /opt/homebrew/bin/node
    npm: 9.2.0 - /opt/homebrew/bin/npm
pnpm: 7.18.2

ct3aMetadata.initVersion: 6.11.2

### Describe the bug

Build of my Next app fails due to missing env variables (I am using the SKIP_ENV_VALIDATION=1 env var) and `Error: Failed to collect page data for /sign/in`

### To reproduce

1. clone https://github.com/vojtechmares/training-platform
2. checkout to branch `ci-docker-build`
3. run `docker build -t platform .` or `SKIP_ENV_VALIDATION=1 pnpm run build`

### Additional information

I believe that the `SKIP_ENV_VALIDATION` env var is not being passed to the app.

If I run the `pnpn run build` outside docker without `/.env` (`/` is project root) file, I get the same error. Altho I was unable to turn off the validation even when I commented out the import in `/next.config.mjs`.

I believe that the environment schema is checked when page data collection starts, since I was able to get this error for different pages.

Unfortunately I was unable to get around this issue.

I am using Next.js version `13.0.6`

**Comment by juliusmarminge** [comment]
Your error isn't from the environment variable validation.

Not sure why it doesn't recognize the custom /sign/in route though...

**Comment by vojtechmares** [comment]
From what I googled, it looks like that Next is trying to run `getServerSideProps` during build time (page collection data?), which results in error.

Unfortunately, I was unable to find out why is this happening and how to fix that. I checked the example from Next documentation for authentication (with NextAuth.js), if I am not missing something and I believe I am not.

I find a discussion at [vercel/next.js](https://github.com/vercel/next.js) (see https://github.com/vercel/next.js/discussions/35534), yet I did not find a solution to this issue.

**Comment by vojtechmares** [comment]
I did some more investigations and googling and I think I am confident in the cause. Just no idea how to fix that.

The `getServerSideProps` method is being called during build time, resulting in chain call up to `unstable_getServerSession`, which requires `authOptions` where is then environment used - which I believe is the reason when providing environment variables, everything passes, but when not, it fails. There is the check and I can't do anything about that, from what I see and know. Yet I am not Next/NextAuth/JS expert.

**Comment by juliusmarminge** [comment]
cc @c-ehrlich @ajcwebdev You have more Docker experience than I have. Any ideas?

**Comment by c-ehrlich** [comment]
The gssp => env vars theory sounds plausible. Will try to take a closer look at this sometime soon but might end up being after Christmas.

**Comment by vojtechmares** [comment]
@c-ehrlich Ok, thank you. Completely understood, happy holidays 🎄

**Comment by vojtechmares** [comment]
reminder cc @c-ehrlich

Can you please have a look? Thank you

**Comment by sebakocz** [comment]
hey I stumbled across this thread because of invalid NEXTAUTH_URL on my static generated pages when deploying to digital ocean

what worked for my was changing 'mywebside.com' to 'http://www.mywebside.com' in my env variables

**Comment by vojtechmares** [comment]
@sebakocz well I know for sure that the value of `NEXTAUTH_URL` is invalid, since during build time the env validation should be skipped and not validated and I want to deploy this as a container to Kubernetes, where `NEXTAUTH_URL` will be provided at runtime.

Your solution does not solve my problem. Thanks tho.

**Comment by c-ehrlich** [comment]
Sorry for the delay. I was able to reproduce your issue and then got it working including GSSP. The reason it was breaking is that we were only checking for `SKIP_ENV_VALIDATION` in `next.config.cjs` but not in the schema files, which are called directly when building routes with GSSP.

Here's the env schema file of my working repo: https://github.com/c-ehrlich/docker-next-auth/blob/main/src/env.mjs - it's a slightly modified version of what Julius is working on in https://github.com/t3-oss/create-t3-app/pull/1092

Hope it's ok that I built it from scratch instead of using your project, but I'm fairly sure it covers the exact thing you're running into (I copy pasted some bits from your project). Let me know if it works for you :)
